File: src/brs/db/store/AssetStore.java
package brs.db.store;

import brs.Asset;
import brs.db.SignumKey;
import brs.db.sql.EntitySqlTable;

import java.util.Collection;

public interface AssetStore {
  SignumKey.LongKeyFactory<Asset> getAssetDbKeyFactory();

  EntitySqlTable<Asset> getAssetTable();

  Collection<Asset> getAssetsIssuedBy(long accountId, int from, int to);

  Asset getAsset(long assetId);

  Collection<Asset> getAssetsByName(String name, int from, int to);
}

File: src/main/kotlin/brs/db/store/AssetStore.kt
package brs.db.store

import brs.entity.Asset
import brs.db.BurstKey
import brs.db.sql.EntitySqlTable

interface AssetStore {
    val assetDbKeyFactory: BurstKey.LongKeyFactory<Asset>

    val assetTable: EntitySqlTable<Asset>

    fun getAssetsIssuedBy(accountId: Long, from: Int, to: Int): Collection<Asset>
}
